### DISCUSSION TITLE
Setting time in AdGuard Home

### DISCUSSION THREAD

**Discussion by ali27868** [AdguardTeam/AdGuardHome#3941] [Q&A] [answered]
Hi!

How does AdGuard Home update its time (for Query Log)? On my router it seems to pull UTC time rather than local time. How can one change this?

Thank you.

**Comment by ainar-g** [comment, marked as answer]
Hello.

If you mean the query log page in the browser, it uses your browser's time zone.

**Reply by ali27868** [reply]
Yes, I meant query log in the browser.

As counterintuitive as it is, but different browsers indeed show different times on the query log: Firefox is UTC, Chrome is local. I always thought that browsers should be using the system time (local time).

**Reply by Chrisvdw00** [reply]
And what about the time in the graph of the DNS queries and filters on the dashboard? it seems one hour off of local time.
 query log for me seems to be in line with local time

**Comment by don-dolarson** [comment]
I'm also concerned about time setting and what could be wrong with it here? AdGuard Home installed on a router.

Router output correct time:
_root@OpenWrt:~# date
Mon May 22 16:20:34 CEST 2023_

System time synchronized ok - a Linux and a Windows machine.
Firefox and Brave on Windows show correct time in AdGuard Home web interface, LibreWolf on Windows and Linux show  -2 hours. System log on router output correct time but a separate adguard.log is also -2 hours.

**Reply by ghost** [reply]
Isn't Resist Fingerprint on LibreWolf active?
Maybe that's the cause.

https://librewolf.net/docs/faq/#what-are-the-most-common-downsides-of-rfp-resist-fingerprinting

https://github.com/arkenfox/user.js/wiki/3.3-Overrides-%5BTo-RFP-or-Not%5D#-rfp

**Reply by don-dolarson** [reply]
I figured it out by goolaG it and that's the case. Trying to figure out if it's possible to skip local addresses from fingerprinting? I mean some exclude list, maybe? Disabling canvas per page doesn't seem to work.

However, what's about AdGuard logs? They're off 2 hours too, and has nothing to do with fingerprinting.

**Reply by ghost** [reply]
> Trying to figure out if it's possible to skip local addresses from fingerprinting? I mean some exclude list, maybe? Disabling canvas per page doesn't seem to work.
>

Nope.
[It is an all-or-nothing buy in: you cannot pick and choose what parts you want.](https://github.com/arkenfox/user.js/blob/04e6e77439bfa6e3f6b7b9c5e0afac7f74f0586a/user.js#L760)

> However, what's about AdGuard logs? They're off 2 hours too, and has nothing to do with fingerprinting.
>

Disable `privacy.resistFingerprinting` then you should be fine.

Tested with `Firefox with arkenfox` and `Mull browser with built in arkenfox`.

And don't forget to install [CanvasBlocker](https://addons.mozilla.org/firefox/addon/canvasblocker/) after disabling resistFingerprinting

**Reply by don-dolarson** [reply]
Thank you. It actually works per-domain basis after some researching and testing.
Tested on Firefox 113.0.1 x64 on Windows 10 22H2 and LibreWolf 113.0.1 on Linux Mint 21.1 x64
http://forums.mozillazine.org/viewtopic.php?f=38&t=3084798

privacy.resistFingerprinting=true
privacy.resistFingerprinting.exemptedDomains=192.168.1.1, http://whatismytimezone.com
privacy.resistFingerprinting.testGranularityMask=4

Note that IP addresses **must** be without http:// or https:// otherwise it won't work. Domains can be with or without it.



The another problem with router logs being off 2 hours is related to adguardhome package. Nothing I can do about.
https://github.com/openwrt/packages/issues/21201
